### PR TITLE
Add revert command

### LINKED ISSUE #59: Add Revert command
As an administrator, I want to be able to undo allocations, so that I can undo the allocations if I made a mistake (e.g., a policy change or emergency).

- [x] Add `revert` command

### PR DESCRIPTION
**Pull request by isaacsaw25**
Added `revert` command that sets the student states back to original unallocated state. This allows the user to reverse the allocation process, edit the list or threshold values, then run the allocation algorithm again.

Close #59

### CODE REVIEW

File: src/main/java/command/RevertCommand.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package command;
+
+import studentlist.StudentList;
+import ui.Messages;
+import ui.UI;
+
+/**
+ * Command to revert the allocation status of all students in the StudentList.
+ * Resets each student's allocation status and associated university index
```
Comment (ehz0ah) [inline src/main/java/command/RevertCommand.java:12]: Good documentation!

File: src/main/java/studentlist/StudentList.java [hunk 2]
```diff
@@ -553,4 +553,13 @@ public ArrayList<Integer> validatePreferences(String preferencesData, Set<String
 
         return preferences;
     }
+
+    /**
+     * Reverts the allocation status of all elements in StudentList.
+     */
+    public void revertAllocation() {
+        for (Student student : students) {
+            student.resetAllocationStatus();
```
Comment (ehz0ah) [inline src/main/java/studentlist/StudentList.java:562]: Nice refactoring to ensure SLAP.

File: src/test/java/parser/ParserTest.java [hunk 3]
```diff
@@ -178,6 +179,41 @@ public void testViewQuotaCommand() {
         System.setOut(originalOut);
     }
 
+    @Test
+    public void testRevertCommand() {
```
Comment (ehz0ah) [inline src/test/java/parser/ParserTest.java:183]: Covered the most important equivalence partition for the method. Good job!

**Review by ehz0ah — APPROVED** [review]
LGTM!

**Comment by ehz0ah** [comment]
Seems like your CICD is failing due to checkstyle errors. Static variable definition is in wrong order under RevertCommand.java

**Comment by isaacsaw25** [comment]
> Seems like your CICD is failing due to checkstyle errors. Static variable definition is in wrong order under RevertCommand.java

Oopsies

**Review by ehz0ah — APPROVED** [review]
ACK

**Pull request merged**